### ISSUE TITLE
"find", "each" method throw type mismatch error.

### ISSUE THREAD

**Issue by Bhagyashree-B** [jducoeur/jquery-facade#11]
I am trying to iterate over options using `val x = $(selectState).find("option").each((y: Element) => y)`. However Compilation throw following error.

```
type mismatch;
 found   : org.scalajs.dom.Element => org.scalajs.dom.Element
    (which expands to)  org.scalajs.dom.raw.Element => org.scalajs.dom.raw.Element
 required: scala.scalajs.js.|[scala.scalajs.js.ThisFunction0[org.scalajs.dom.Element,Any],scala.scalajs.js.ThisFunction1[org.scalajs.dom.Element,Int,Any]]
    (which expands to)  scala.scalajs.js.|[scala.scalajs.js.ThisFunction0[org.scalajs.dom.raw.Element,Any],scala.scalajs.js.ThisFunction1[org.scalajs.dom.raw.Element,Int,Any]]
```

I was referring to stack overflow [question](http://stackoverflow.com/questions/33483226/call-jquery-each-method-from-scala-js). Is it changed in recent library?

**Comment by gzm0** [comment]
Chipping in since I saw this on SO:

I think in terms of type inference and implicit conversion, you are better of defining overloads in such a case. AFAICS, this should work here, since the erased types are different.

So I'd try something like this:

``` scala
def each(func:js.ThisFunction0[Element, Any]):JQuery = js.native
def each(func:js.ThisFunction1[Element, Int, Any]):JQuery = js.native
```

**Comment by jducoeur** [comment]
Yep -- have already checked that change in, and am waiting for the new release to propagate before I close this all out...

**Comment by jducoeur** [comment]
Okay, fixed in 1.0-RC3.
